### DISCUSSION TITLE
Make LDAP certificate optional for public trusted CA

### DISCUSSION THREAD

**Discussion by fwPunsher** [open-webui/open-webui#7343] [Ideas]
When connecting against an ldap server that has a certificate that has been issued by a public trusted CA the certificate field should be optional and the connection should be possible without specifying any certificate.

Background: we are issuing our ldap servers with certificates that are signed via Lets Encrypt. As such the application should simply accept the certificate without any further certificate configuration necessary.

**Comment by ips972** [comment]
+1 - please add the option to allow for secure ldap without loading a certificate. i know its less secure, but its a close network. and i trust the AD. - tnx!

**Comment by soppelmann** [comment]
Should be as simple as having a setting for this line 
https://github.com/open-webui/open-webui/blob/c4ea31357f49d08a14c86b2bd85fdcd489512e91/backend/open_webui/apps/webui/routers/auths.py#L186
to toggle between CERT_NONE and CERT_REQUIRED from ldap3 documentation at 
https://ldap3.readthedocs.io/en/latest/ssltls.html

**Reply by fwPunsher** [reply]
To completely skip the validation yes. However I would rather suggest to rely on the default set of known CAs. One example would be this set: https://pypi.org/project/certifi/
